pkg/oci/oci.go

```diff
@@ -180,15 +177,31 @@ func pullIfNotExist(ctx context.Context, imageStore oras.Target, authOpts *AuthO
 		return fmt.Errorf("resolving image %q: %w", image, err)
 	}
 
+	_, err = pullImage(ctx, targetImage, imageStore, authOpts)
+	if err != nil {
+		return fmt.Errorf("pulling image %q: %w", image, err)
+	}
+
+	return nil
+}
+
+func pullImage(ctx context.Context, targetImage reference.Named, imageStore oras.Target, authOpts *AuthOptions) (*ocispec.Descriptor, error) {
+	if authOpts.DisablePulling {
+		return nil, errors.New("pulling is disabled due to using corresponding option")
```

Comment:
I'm not sure how to improve this message. Some ideas:

```suggestion
		return nil, errors.New("pulling is disabled in the configuration")
```

or just 


```suggestion
		return nil, errors.New("pulling is disabled")
```